src/block-cursor.ts

```diff
@@ -158,11 +158,15 @@ function measureCursor(cm: CodeMirror, view: EditorView, cursor: SelectionRange,
       domAtPos = {node: domAtPos.node.childNodes[domAtPos.offset], offset: 0};
     }
     if (!(node instanceof HTMLElement)) {
+      if (!node.parentNode) return null;
       node = node.parentNode;
     }
     let style = getComputedStyle(node as HTMLElement);
     let letter = head < view.state.doc.length && view.state.sliceDoc(head, head + 1);
     if (!letter || letter == "\n" || letter == "\r") letter = "\xa0";
+    else if ((/[\uD800-\uDC00]/.test(letter) && head < view.state.doc.length - 1)) {
```

Comment:
can we add surrogate pair comment here too?

Reply:
reading the [doc](https://learn.microsoft.com/en-us/globalization/encoding/surrogate-pairs) here, high surrogate ranges from U+D800 to U+DBFF. should this include `\uDC00`?